static/js/todo_widget.js

```diff
@@ -0,0 +1,196 @@
+var todo_widget = (function () {
+
+var exports = {};
+
+exports.task_data_holder = function (is_my_task_list) {
+    var self = {};
+
+    var all_tasks = [];
+    var pending_tasks = [];
+    var completed_tasks = [];
+    var my_idx = 0;
+
+    self.get_widget_data = function () {
+
+        var widget_data = {
+            pending_tasks: pending_tasks,
+            completed_tasks: completed_tasks,
+        };
+
+        return widget_data;
+    };
+
+    self.check_task = {
+        task_exists: function (task) {
+            var task_exists = false;
+            all_tasks.forEach(function (item) {
+                if (item.task === task) {
```

Comment:
This code looks fine, but I recommend that you add `_.any(...)` to your bag of tricks, and replace this with the code below.  The `_.any()` function is one of many useful functions in the underscore library:  https://underscorejs.org/

~~~ js
var task_exists = _.any(all_tasks, function (item) {
   return (item.task === task);
});